pkg/operator/controller/gatewayclass/controller.go

```diff
@@ -97,6 +136,9 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	if _, _, err := r.ensureServiceMeshOperatorSubscription(ctx); err != nil {
 		errs = append(errs, err)
 	}
+	if _, _, err := r.ensureServiceMeshOperatorInstallPlan(ctx); err != nil {
```

Comment:
Please add a unit test for this function, or the Reconcile function, using the pattern in the gatewayapi/controller_test.go if possible.  At least make sure the install plan is not created if it's missing, and that it is properly reconciled if changes are attempted that are not allowed.